lib/Backend/Lower.cpp

```diff
@@ -5710,9 +5710,6 @@ Lowerer::GenerateLdFldWithCachedType(IR::Instr * instrLdFld, bool* continueAsHel
 
     if (hasTypeCheckBailout)
     {
-        AssertMsg(PHASE_ON1(Js::ObjTypeSpecIsolatedFldOpsWithBailOutPhase) || !propertySymOpnd->IsTypeDead(),
```

Comment:
Turned out that it is fine to leave assert here and the one below because we will only come this code path if there is no typeCheckBailout.
```c++
if(hasTypeCheckBailout) {

```